src/averageAges.js

```diff
@@ -55,7 +65,18 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childs = onlyWithSon
+    ? people.filter(person => people.some(male =>
+      male.name === person.mother && person.sex === 'm'))
+    : people.filter(person => people.some(male => male.name === person.mother));
```

Comment:
```suggestion
  const childs= people.filter(person=> onlyWithSon
    ? person.sex === 'm' && people.some(mother => mother.name === person.mother)
    : people.some(mother => mother.name === person.mother));
```

Same but a bit shorter